File: backendcloud/sharingsystem/src/main/java/com/ticket/sharingsystem/pojo/Sharing.java
package com.ticket.sharingsystem.pojo;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document("sharing")
/**
 * @author: VenusHui
 * @description: TODO
 * @date: 2022/12/27 22:54
 * @version: 1.0
 */
public class Sharing {
    @Id
    private String _id;
    private Integer histrionicsId;
    private Integer userId;
    private String context;
    private String imageUrl;
    private Integer stars;
    private Integer likes;
    private Date shareDate;
    private List<Comment> comments;
}

File: frontend/OneOnly/lib/models/sharing.dart
class SharListResponse {
  List<Shar>? content;

  SharListResponse({
    this.content,
  });

  factory SharListResponse.fromJson(Map<String, dynamic> json) =>
      SharListResponse(
        content: json["content"] == null
            ? []
            : List<Shar>.from(json["content"].map((x) => Shar.fromJson(x))),
      );

  Map<String, dynamic> toJson() => {
        "content": content == null
            ? []
            : List<dynamic>.from(content!.map((x) => x.toJson())),
      };
}

class Shar {
  String? id;
  int? histrionicsId;
  int? userId;
  String? context;
  String? imageUrl;
  int? stars;
  int? likes;
  String? shareDate;
  bool isLike = false;
  Map? shows = {};
  Shar({
    this.id,
    this.histrionicsId,
    this.userId,
    this.context,
    this.imageUrl,
    this.stars,
    this.likes,
    this.shareDate,
    this.shows,
  });

  factory Shar.fromJson(Map<String, dynamic> json) => Shar(
        id: json["_id"],
        histrionicsId: json["histrionicsId"],
        userId: json["userId"],
        context: json["context"],
        imageUrl: json["imageUrl"],
        stars: json["stars"],
        likes: json['likes'],
        shareDate: json['shareDate'],
        shows: json["shows"],
      );

  Map<String, dynamic> toJson() => {
        "_id": id,
        "histrionicsId": histrionicsId,
        "userId": userId,
        "context": context,
        "imageUrl": imageUrl,
        "stars": stars,
        "likes": likes,
        "shows": shows,
      };
}
